src/middle/Semantic_error.ml

```diff
@@ -177,19 +180,23 @@ module TypeError = struct
             (with explicit types for variadic args). *)
         let variadic_ode_generic_signature =
           let optional_tol_args =
-            if Stan_math_signatures.is_variadic_ode_tol_fn name then
-              types Stan_math_signatures.variadic_ode_tol_arg_types
+            if Stan_math_signatures.variadic_ode_adjoint_fn = name then
+              types Stan_math_signatures.variadic_ode_adjoint_ctl_tol_arg_types
+            else if Stan_math_signatures.is_nonadjoint_variadic_ode_tol_fn name
+            then types Stan_math_signatures.variadic_ode_tol_arg_types
             else []
           in
           match
-            ( types Stan_math_signatures.variadic_ode_mandatory_arg_types
-            , types Stan_math_signatures.variadic_ode_mandatory_fun_args )
+            ( Stan_math_signatures.variadic_ode_mandatory_arg_types
+            , Stan_math_signatures.variadic_ode_mandatory_fun_args )
           with
           | arg0 :: arg1 :: arg2 :: _, fun_arg0 :: fun_arg1 :: _ ->
               Fmt.strf "(%a, %a, ...) => %a, %a, %a, %a, %a ...\n"
-                UnsizedType.pp fun_arg0 UnsizedType.pp fun_arg1 UnsizedType.pp
+                UnsizedType.pp_fun_arg fun_arg0 UnsizedType.pp_fun_arg fun_arg1
+                UnsizedType.pp
                 Stan_math_signatures.variadic_ode_fun_return_type
-                UnsizedType.pp arg0 UnsizedType.pp arg1 UnsizedType.pp arg2
+                UnsizedType.pp_fun_arg arg0 UnsizedType.pp_fun_arg arg1
+                UnsizedType.pp_fun_arg arg2
```

Comment:
This change is so the error also prints `data` for all variadic ODE types.

Reply:
IMO the signatures are pretty much unreadable either way but I suppose the data markers help if that was what caused the error.

Reply:
Yeah, I agree that the prints of the two large signatures is not great... But that is a different issue I guess as its not limited to just the variadic ODEs.